packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2078,12 +2078,27 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     _computeTextMetricsIfNeeded();
     final TextPosition firstPosition = _textPainter.getPositionForOffset(globalToLocal(from - _paintOffset));
     final TextSelection firstWord = _getWordAtOffset(firstPosition);
-    final TextSelection lastWord = to == null ?
-      firstWord : _getWordAtOffset(_textPainter.getPositionForOffset(globalToLocal(to - _paintOffset)));
+    final TextPosition lastPosition = to == null ? firstPosition : _textPainter.getPositionForOffset(globalToLocal(to - _paintOffset));
+    final TextSelection lastWord = lastPosition == firstPosition ? firstWord : _getWordAtOffset(lastPosition);
+
+    final bool isFirstWordBeforeLast = firstPosition.offset < lastPosition.offset;
+    final bool firstWordIsLast = firstWord == lastWord;
```

Comment:
nit: probably don't need this?

Reply:
also consider renaming first -> from, last -> to, to match the names of the args.

Reply:
Do you mean the conditionals can just be in line?

```dart
if (fromWord == toWord) {
    
} else if (fromPosition .offset < toPosition.offset) {

}
```

Reply:
I meant `firstWord.start > lastWord.end ? (firstWord.end, lastWord.start) : (firstWord.start, lastWord.end)`. Didn't notice you were comparing `firstPosition` and `lastPosition`.

Reply:
Still a little confused.

For `also consider renaming first -> from, last -> to, to match the names of the args.` Did you mean change `firstPosition` -> `fromPosition`, `lastPosition` -> `toPosition` or change `firstWord` -> `fromWord`, `lastWord` -> `toWord`?

Reply:
Also what line where you referring to for `nit: probably don't need this?`.

Reply:
I meant a `s/first/from/` etc. first/last are a bit ambiguous since it's unclear what order it's referring to.

 > Also what line where you referring to for nit: probably don't need this?

Special casing `firstWordIsLast` can be avoided I think. But it's not a big deal.

Reply:
I think it could as well but some tests were failing because without it the case always default to false if `from == to` which means the `firstWord.extent.offset` will be the base and `lastWord.base.offset` will be the extent. I think i'll leave it for now to keep those tests happy.

Reply:
Are you comparing `fromPositoin.offset` and `toPosition.offset`? firstWord.start > lastWord.end ? (firstWord.end, lastWord.start) : (firstWord.start, lastWord.end) should work?

Reply:
Ahh you're right that worked and cleans up the code a bit. Thanks!